kong/include/kong/services/negotiation/v1/negotiation.proto

```diff
@@ -0,0 +1,58 @@
+syntax = "proto3";
+
+option go_package = "github.com/kong/koko/internal/gen/wrpc/kong/model;model";
+
+package kong.model;
+
+
+// NegotiationService allows DP to ask for services and CP answer with the accepted version of each
+// +wrpc:service-id=5
+service NegotiationService {
+  // NegotiateServices should be the first RPC call upon connecting.  It allows the CP
+  // to get which services does the DP handle.
+  //
+  // Call direction: DP to CP
+  // +wrpc:rpc-id=4
+  rpc NegotiateServices(kong.model.NegotiateServicesRequest) returns (kong.model.NegotiateServicesResponse);
+}
+
+message NegotiateServicesRequest {
+  DPNodeDescription node = 1;
+  map<string, ServiceVersions> services_requested = 2;
+}
+
+message DPNodeDescription {
+  string type = 2;
+  string version = 3;
+  string hostname = 4;
+}
+
+// When version is undefined, service is denied.
+message ServiceVersion {
+  string version = 1;
+  string description = 2;
+}
+
+message ServiceVersions {
+  repeated ServiceVersion service = 1;
```

Comment:
the request don't have to push the description, just the versions it supports is fine.  i see the code doesn't put the description, so it wouldn't use any space, but removing it allows you to use a `repeated string`.

also, as you mentioned, protobuf doesn't allow a map to array, but an array of sub messages is much simpler both in proto definition and in the handling code:

```
message NegotiateServicesRequest {
  DPNodeDescription node = 1;
  repeated ServiceRequest services_requested = 2;
}

message ServiceRequest {
  string name = 1;
  repeated string versions = 2;
}
```